aptos-move/move-examples/token_objects/sources/hero.move

```diff
@@ -0,0 +1,337 @@
+module token_objects::example {
+    use std::option::{Self, Option};
+    use std::signer;
+    use std::string::{Self, String};
+
+    use aptos_framework::object::{
+        Self,
+        CreatorAbility,
+        DeleteAbility,
+        OwnerAbility,
+        TypedOwnerAbility,
+    };
+
+    use token_objects::token;
+
+    const ENOT_A_HERO: u64 = 1;
+    const ENOT_A_WEAPON: u64 = 2;
+    const ENOT_A_GEM: u64 = 3;
+    const ENOT_CREATOR: u64 = 4;
+
+    struct OnChainConfig has key {
+        collection: String,
+        mutability_config: token::MutabilityConfig,
+        royalty: token::Royalty,
+    }
+
+    #[resource_group(container = aptos_framework::object::ObjectGroup)]
```

Comment:
What's the benefit of this pattern of grouping everything into ObjectGroup? It seems strange as this would encourage everyone to group their objects into the same resource group, making it impossibly large.